### ISSUE TITLE
deserialize method is vulnerable

### ISSUE THREAD

**Issue by Kay-MalwareBenchmark** [commenthol/serialize-to-js#3]
![wx20171023-202047](https://user-images.githubusercontent.com/33020407/31888720-bde8ed54-b82f-11e7-85d0-cd83a92a7131.png)
Hi, there is a vulnerability in deserialize method, please see PoC above.

**Comment by commenthol** [comment]
Can you please explain what your payload tries to exploit in detail? 
In my understanding it just runs inside a "sandbox" and prints "exploit" to a tty. 
Please check https://github.com/commenthol/safer-eval/blob/master/test/saferEval.js#L61
There you can find other tests which try preventing exploits. 
Please rise you issue(s) there.

**Comment by Kay-MalwareBenchmark** [comment]
Thx

**Comment by Kay-MalwareBenchmark** [comment]
Hi, I have read the Node.js doc, and comfirm that deserialize method can not be used to RCE which can be harmful to server. But it  may cause denial of service attacks, if attackers inject endless loops to deserialize method.
What do you think?

**Comment by commenthol** [comment]
It is as it is. Any tool can be misused. Yes you are right a simple "while(1) {}" loop blocks the main thread. DoS done.
The module is named "serialize-to-js" so deserialization is not the main intented usecase. The README contains a big warning stating that the `deserialize` method might be harmful. [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval$edit#Don%27t_use_eval_needlessly!) also states something similar but is not removed from the standard as there might be usecases which do require this kind of method. So someone could wonder why there is still no CVE filed on`eval`? 
So if you think that things are still not clear enough I am open to suggestions. But I still think that leaving the method inside is still a better option than letting a novice choose `eval`.

**Comment by Kay-MalwareBenchmark** [comment]
Yeah, maybe there is no available solution for this vulnerability, and the only way to avoid potential risk is not use it in network applications in combination with untrusted user input.

**Comment by Noiwex** [comment]
https://opsecx.com/index.php/2017/02/08/exploiting-node-js-deserialization-bug-for-remote-code-execution/
